source/standard_scores.Rmd

```diff
@@ -0,0 +1,1523 @@
+whatever you send it.  Here we select the first six rows of the data frame.
+
+```{r}
+# Show the first six rows in the data frame
+head(district_income)
+```
+
+The data are in income order, sorted lowest to highest, so the first five
+districts are those with the lowest household income.
+
+We are particularly interested in the column named `Median_Income`.
+
+You can fetch columns of data from a data frame by using R's `$` syntax.  The
+`$` syntax means "fetch the thing named on the right of the `$` attached to the
+value given to the left of the `$`".
+
+So, to get the data for the `Median_Income` column, we can write:
+
+```{r}
+# Use $ syntax to get a column of data from a data frame.
+# "fetch the Median_Income thing from district_income".
+incomes = district_income$Median_Income
+# The thing that comes back is our familiar R vector.
+# Show the first five values, by indexing with a slice.
+incomes[1:5]
+```
+
+:::
+
+### Incomes and Ranks
+
+We now have the `incomes` values as {{< var an_array >}}.
+
+There are 441 values in the whole vector, one of each congressional
+district:
+
+```{python}
+len(incomes)
+```
+
+```{r}
+length(incomes)
+```
+
+While we are at it, let us also get the values from the "Ascending_Rank"
+column, with the same procedure.  These are ranks from low to high, meaning
+1 is the lowest median income, and 441 is the highest median income.
+
+```{python}
+lo_to_hi_ranks = np.array(district_income['Ascending_Rank'])
+# Show the first five values, by indexing with a slice.
+lo_to_hi_ranks[:5]
+```
+
+```{r}
+lo_to_hi_ranks <- district_income$Ascending_Rank
+# Show the first five values, by indexing with a slice.
+lo_to_hi_ranks[1:5]
+```
+
+In our case, the [data frame]{.r}[DataFrame]{.python} has the `Ascending_Rank`
+column with the ranks we need, but if we need the ranks and we don't have them,
+we can calculate them using the [`rank` function]{.r}[`rankdata` function from
+the Scipy `stats` package]{.python}.
+
+::: python
+
+### Introducing Scipy
+
+Earlier in this chapter we introduced the Pandas module.  We used Pandas to
+load the CSV data into Python.
+
+Now we introduce another fundamental Python library for working with data
+called Scipy.  The name Scipy comes from the compression of SCIentific PYthon,
+and the library is nearly as broad as the name suggests — it is a huge
+collection of functions and data that implement a wide range of scientific
+algorithms.  Scipy is an umbrella package, in that it contains sub-packages,
+each covering a particular field of scientific computing.  One of those
+sub-packages is called `stats`, and, yes, it covers statistics.
+
+We can get the Scipy `stats` sub-package with:
+
+```{python}
+import scipy.stats
+```
+
+but, as for Numpy and Pandas, we often `import` the package with an
+abbreviation, such as:
+
+```{python}
+# Import the scipy.stats package with the name "sps".
+import scipy.stats as sps
```

Comment:
Or import scipy as sp and use sp.stats.whatever

Reply:
Several versions of SciPy at least. The problem is that you otherwise have to make up abbreviations for each submodule import, and this will eventually confuse the reader. But, again no super strong feelings here.  Perhaps `from scipy import stats` may be the tidiest option?